src/main.ts

```diff
@@ -13,18 +13,33 @@ export async function run(): Promise<void> {
     const secrets: string[] = core.getMultilineInput("secrets", {
       required: true,
     });
+    const cloudRegion = core.getInput("cloud_region");
     const baseUrl: string = core.getInput("base_url");
     let identityUrl: string = core.getInput("identity_url");
     let apiUrl: string = core.getInput("api_url");
 
     core.info("Validating bitwarden/sm-action inputs...");
-    if (baseUrl) {
-      if (!isValidUrl(baseUrl)) {
-        throw TypeError("input provided for base_url not in expected format");
+    if (identityUrl || apiUrl) {
+      if (baseUrl) {
+        if (!isValidUrl(baseUrl)) {
+          throw TypeError("input provided for base_url not in expected format");
+        }
+        identityUrl = baseUrl + identityUrl;
+        apiUrl = baseUrl + apiUrl;
```

Comment:
With this change, setting just the base_url will no longer work.

For self-hosted, the expectation is that you either set the base_url or pair of identity_url and api_url - as documented in README.md

Reply:
@mzieniukbw @kimdre with the new addition of another Parameter in the Readme, coupled with the code we currently have with the new addition of cloudRegion, I found that it is getting pretty confusing for someone who hasn't done this before. I suggest that we update the ReadMe to a format like so: 
![image](https://github.com/bitwarden/sm-action/assets/106776772/7a58efe8-cb49-4ded-828c-840d25945e13)

As for the code, I have played a little bit around with rewriting the code to make it more understandable (hopefully), and I'd like to hear both of your thoughts.

Instead of something like this: 
`  if (baseUrl) {
      if (!isValidUrl(baseUrl)) {
        throw TypeError("input provided for base_url not in expected format");
      }
      identityUrl = baseUrl + "/identity";
      apiUrl = baseUrl + "/api";
    } else {
      if (!identityUrl && !apiUrl) {
        let cloudBaseUrl;
        if (cloudRegion === "us") {
          cloudBaseUrl = "bitwarden.com";
        } else if (cloudRegion === "eu") {
          cloudBaseUrl = "bitwarden.eu";
        } else {
          throw TypeError("input provided for cloud_region not in expected format");
        }
        identityUrl = "https://identity." + cloudBaseUrl;
        apiUrl = "https://api." + cloudBaseUrl;
      }
    }
`
We should rewrite it to something like this:
`
let customUrls;
//Check if only identityUrl is set and not apiUrl: if so throw an error, otherwise mark this as the user is using customUrls  
if(identityUrl||apiUrl){
  //If either are set, make sure both are.
  if(identityUrl && apiUrl){
    throw TypeError("If using custom Urls, Both identityUrl and apiUrl need to be set.");
  }
  customUrls= true;
}

var selfHosted  = baseUrl || customUrls;
if(selfHosted){ //Self Hosted
  //if self hosted cloudBaseUrl will have no impact, but customUrls will have an impact.
  if(baseUrl && customUrls){
    throw TypeError("Please use baseUrl OR apiUrl/identityUrl, you cannot use both.");
  }
  if(baseUrl){
    identityUrl = baseUrl + "/identity";
    apiUrl = baseUrl + "/api";
  }
} else { //BW Hosted
  //Bw hosted, allows users to set cloudRegion, by default this value is "us"
  //if it is Bw hosted there should not be an identityUrl or an apiUrl specified - because we will overwrite it with the bw hosted defaults of https://identity.bitwarden.com or .eu
  if(!customUrls){ 
    let cloudBaseUrl;
    if (cloudRegion === "us") {
          cloudBaseUrl = "bitwarden.com";
        } else if (cloudRegion === "eu") {
          cloudBaseUrl = "bitwarden.eu";
        } else {
          throw TypeError("input provided for cloud_region not in expected format");
        }
        identityUrl = "https://identity." + cloudBaseUrl;
        apiUrl = "https://api." + cloudBaseUrl;
  } else {
    throw TypeError("We have detected you have decided to use Bitwarden hosting instead of self hosting, but you've defined identityUrl or apiUrl, these values should not be defined if you are using Bitwarden hosting.");
  }
}`

**The code above needs cleaning up and to be tested locally**

@kimdre I'm very happy you created this community PR as it has us looking into improving this code/readme, and that's awesome! Please let me know how you would like to proceed, if you'd like to work on it yourself that's great, but if you'd like a hand I can commit changes to your fork.

Reply:
@mzieniukbw how do you feel about this suggestion? I know you have suggested erroring on cases where the user is setting more variables than needed, however I am concerned with backwards compatibility. Changing it to now error would result in every user who accidentally set a parameter that wasn't needed to have to go edit their parameters to remove it - where in the past it just ignored that issue. I assume we want to keep it backwards compatible, and in that case we should remove the lines of code where I am throwing an error and instead just send a message to the user and continue on like it did in the past.

Reply:
@cd-bitwarden sure that works 👍